### PR TITLE
Initial proto definition

### PR DESCRIPTION
**Pull request by rimrakhimov**
User story: #201

First iteration of Ethereum Bytecode Database service API. Creates proto file with Grpc and Http transcoded API definitions.

### CODE REVIEW

File: ethereum-bytecode-database-proto/ethereum-bytecode-database-proto/proto/ethereum-bytecode-database.proto [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+service CreationBytecodeDatabase {
+  rpc VerifySolidityMultiPart(VerifySolidityMultiPartRequest) returns (VerifyResponse) {}
+
+  rpc VerifySolidityStandardJson(VerifySolidityStandardJsonRequest) returns (VerifyResponse) {}
+
+  rpc VerifySolidityMetadata(VerifySolidityMetadataRequest) returns (VerifyResponse) {}
+
+  rpc VerifyVyperMultiPart(VerifyVyperMultiPartRequest) returns (VerifyResponse) {}
+
+  rpc SearchSources(SearchSourcesRequest) returns (SearchSourcesResponse) {}
```
Comment (leviska) [inline ethereum-bytecode-database-proto/ethereum-bytecode-database-proto/proto/ethereum-bytecode-database.proto:16]: why is it here too?

File: ethereum-bytecode-database-proto/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[workspace]
+members = ["ethereum-bytecode-database-proto"]
```
Comment (sevenzing) [inline ethereum-bytecode-database-proto/Cargo.toml:2]: why is main folder for this service called `ethereum-bytecode-database-proto` ? maybe remove `proto`? I don't understand why you repeat it twice

File: ethereum-bytecode-database-proto/ethereum-bytecode-database-proto/proto/ethereum-bytecode-database.proto [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  /// Status of 0 indicates successful verification
+  int32 status = 2;
+
+  message Result {
+    /// The name of the file verified contract was located at
+    string file_name = 1;
+    /// The name of the contract which was verified
+    string contract_name = 2;
+    /// A source used for the verification
+    Source source = 4;
```
Comment (sevenzing) [inline ethereum-bytecode-database-proto/ethereum-bytecode-database-proto/proto/ethereum-bytecode-database.proto:133]: still don't understand why blockscout needs source files. Like, it needs compiler version, filename, etc but not content of files. files may be really big, so it looks wasteful.

File: ethereum-bytecode-database-proto/ethereum-bytecode-database-proto/proto/ethereum-bytecode-database.proto [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+message Source {
+  // Compiler version used to compile the contract
+  string compiler_version = 1;
+  /// 'settings' key in Standard Input JSON
+  /// (https://docs.soliditylang.org/en/latest/using-the-compiler.html#input-description)
+  string compiler_settings = 2;
+
+  enum Match {
+    MATCH_UNSPECIFIED = 0;
```
Comment (sevenzing) [inline ethereum-bytecode-database-proto/ethereum-bytecode-database-proto/proto/ethereum-bytecode-database.proto:43]: does sourcify have unspecified match type? what is a situation when we could have this match type?

Comment (rimrakhimov) [reply ethereum-bytecode-database-proto/ethereum-bytecode-database-proto/proto/ethereum-bytecode-database.proto]: Sourcify does not have unspecified match type, but Grpc should have some default enum variant and partial match does not seem to be the actual default type. Adding an 'unknown' may make it more flexible (https://google.aip.dev/126)

**Review by sevenzing — APPROVED** [review]

**Review by leviska — APPROVED** [review]

**Pull request merged**